packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -89,14 +89,12 @@ contract TaikoL1 is EssentialContract, ITaikoL1, TaikoEvents, TaikoErrors {
             }
         }
 
-        require(_params.length == _txLists.length, "mismatched params length");
-
         TaikoData.Config memory config = getConfig();
         TaikoToken tko = TaikoToken(resolve(LibStrings.B_TAIKO_TOKEN, false));
 
-        for (uint256 i = 0; i < _params.length; i++) {
+        for (uint8 i = 0; i < _params.length; i++) {
```

Comment:
If we use `uint8` we should make sure `_params.length` is less than uint8.Max value.

Reply:
good point, I don't think anyone would do 256 blocks at once because it would run out of gas though

Reply:
There is no harm in us thinking about this situation, but when it happens, it is fatal.

Reply:
Logic will be changed and moved outside to a different function and I will change to handle this scenario